Stats: Update MobilePromoCard component to use redirects

a8ck3n opened the pull request:
#### Proposed Changes

Removes the raw URLs in favour of the jetpack.com redirects. This allows us to update the URLs server-side without having to make code changes. Applies to QR codes as well.

#### Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Check out this branch.
* Run `yarn workspace @automattic/components run storybook`.
* Visit the Mobile Promo Card entry.
* Confirm the updated QR codes display correctly.
* Confirm the QR codes link out correctly. (they go via jetpack.com/redirect now)
* Use the Chrome dev tools and change to the "iPhone 12 Pro" setting. Refresh the view and confirm you get an App Store badge that links to the app page.
* Change to the "Pixel 5" setting and refresh again. Confirm you get a Play Store badge and that it's wired up correctly.
* For a12s, visit jetpack.com/redirect to confirm redirects are working.


#### Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Replaces redirect work in #70521. Blocker for #70463.

a8ck3n committed:
Replace all direct URLs with redirects.

a8ck3n committed:
Remove reference URLs.

a8ck3n committed:
Updated QR codes with redirect URLs.

a8ck3n committed:
Use includes instead of indexOf for array test.

a8ck3n committed:
Rename helper function for clarity.

a8ck3n committed:
Refactor to use configuration object.

a8ck3n committed:
Rename variable for accuracy.

jsnmoon reviewed (commented):
The overall approach looks solid, but I think could use some improvements. Here are my observations:

1. The fallback behavior for `getRedirectUrlFromSlug` makes the `redirectLink` for `woo.com/mobile` and `jetpack.com/app` point to the current page via `href=""`.
2. `getRedirectUrlFromSlug`'s input parameter can be simplified to use an identifier (e.g., `'wooApple'`) instead of the actual slug (e.g., `REDIRECT_SLUGS.wooApple`).

You can find my suggested changes [here](https://github.com/Automattic/wp-calypso/compare/update/mobile-promo-use-redirects...update/mobile-promo-use-redirects-feedback); feel free to cherry-pick!

a8ck3n committed:
Inline string interpolation.

a8ck3n committed:
Add fallback URL in case lookup fails.

a8ck3n committed:
Small code flow tweak for readability.

a8ck3n committed:
Refactor getRedirectUrl helper.

Valid result or null instead of returning an empty string. Also updates call sites.

a8ck3n commented:
> The overall approach looks solid, but I think could use some improvements. Here are my observations:

Thanks @jsnmoon! I've added your recommended changes here.

jsnmoon reviewed (approved):
Looks solid. Ship away!
